### ISSUE TITLE
Achieve compatibility with Rocket.Chat

### ISSUE THREAD

**Issue by b1ron** [FerretDB/FerretDB#1993]
### What should be done?

We cannot run their [end-to-end tests](https://developer.rocket.chat/rocket.chat/contribute-to-rocket.chat/code-style-guide#end-to-end-tests) because the server relies on either tailing the oplog or change streams. Disabling both will make the Rocket.Chat instance crash. To disable both set `IGNORE_CHANGE_STREAM=true` and do not set the `MONGO_OPLOG_URL` environment variable.

So below is going through the source code:

* https://github.com/FerretDB/FerretDB/issues/1509
* https://github.com/FerretDB/FerretDB/issues/1413
* https://github.com/FerretDB/FerretDB/issues/1427
* https://github.com/FerretDB/FerretDB/issues/1439
* https://github.com/FerretDB/FerretDB/issues/1442
* https://github.com/FerretDB/FerretDB/issues/1420
* https://github.com/FerretDB/FerretDB/issues/1423
* https://github.com/FerretDB/FerretDB/issues/1425
* https://github.com/FerretDB/FerretDB/issues/1456
* https://github.com/FerretDB/FerretDB/issues/1438
* https://github.com/FerretDB/FerretDB/issues/1467
* https://github.com/FerretDB/FerretDB/issues/1455
* https://github.com/FerretDB/FerretDB/issues/1434
* https://github.com/FerretDB/FerretDB/issues/1436
* https://github.com/FerretDB/FerretDB/issues/1467
* https://github.com/FerretDB/FerretDB/issues/414
* https://github.com/FerretDB/FerretDB/issues/1453 (`$divide`)
* https://github.com/FerretDB/FerretDB/issues/1457 (`$cond`, `$ifNull`)
* https://github.com/FerretDB/FerretDB/issues/1992

**Comment by geekgonecrazy** [comment]
You can replace #76 with #175 Rocket.Chat has done a lot of work to move to changestreams instead of oplog

**Comment by ptrfarkas** [comment]
Thank you for your feedback @geekgonecrazy ! We may have further questions later on this if you don't mind. 
Unfortunately, many other applications are not taking steps to move away from OpLog, so it is unclear whether we will be able to save the hassle of implementing it or not.

**Comment by geekgonecrazy** [comment]
Understood.  Yeah unfortunately meteor tapped into oplog to watch for db changes.  While it did help add that magic reactivity.  It also made it hard for them to now move to the rolling mongo built to solve that need.

Definitely count me in to help get any you need.  Have been watching this project for a while hoping to see it get far enough to be able to use with Rocket.Chat

**Comment by xet7** [comment]
@AlekSi 

Can you also fix this tasklist error?

**Comment by maudrid** [comment]
Considering that FerretDB has a v2 RC out. What is the update here?
I tried using RC with the v2 RC, but got this error:
```
MongoServerError: Stage $changeStream is not supported yet in native pipeline
```
Is the list of tasks in this ticket still valid for V2?

**Comment by AlekSi** [comment]
I guess Rocket.Chat switched from OpLog to Change Streams? https://github.com/microsoft/documentdb/issues/50

**Comment by geekgonecrazy** [comment]
> You can replace https://github.com/FerretDB/FerretDB/issues/76 with https://github.com/FerretDB/FerretDB/issues/175 Rocket.Chat has done a lot of work to move to changestreams instead of oplog

Indeed.  I think still might be possible via an environment variable to use oplog.  But it’s not been the default for a while, changestream has gotten a lot of attention and optimization.  It’s definitely lighter than the oplog implementation
